src/components/favorites-location-items/favourites-location-items.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import PropTypes from "prop-types";
+import FavouritesLocationGroup from "../favourites-location-group/favourites-location-group";
+
+const groupByCity = (arr = []) => {
+  let result = [];
+  result = arr.reduce((r, a) => {
+    r[a.city] = r[a.city] || [];
+    r[a.city].push(a);
+    return r;
+  }, Object.create(null));
```

Comment:
обычно достаточно вот так)
```suggestion
  }, {});
```